Tools/CISupport/Shared/de-duper

```diff
@@ -114,21 +119,28 @@ def main(args):
             if os.stat(full_path).st_nlink > 1:
                 continue
             sha_value = sha1(full_path)
+            sha_path = os.path.join(args.storage, f'{sha_value}.png')
+            if sha_path == full_path:
+                continue
             if sha_value in existing_links:
                 if args.dry_run:
                     print(f'[dry-run] Linking {full_path} to {sha_value}.png')
                 else:
                     os.remove(full_path)
-                    os.link(os.path.join(args.storage, f'{sha_value}.png'), full_path)
+                    os.link(sha_path, full_path)
```

Comment:
I do still wonder if we want to do something like:

```
try:
    os.link(sha_path, full_path)
except OSError as e:
    if e.errno == errno.EMLINK:
        os.remove(sha_path)
        os.link(sha_path, full_path)
    else:
        raise
```

To guard against other processes also creating links?